aipmiddleware/validateparent_test.go

```diff
@@ -0,0 +1,166 @@
+package aipmiddleware
+
+import (
+	"context"
+	"net"
+	"sync"
+	"testing"
+
+	examplefreightv1 "go.einride.tech/aip/examples/proto/gen/einride/example/freight/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/testing/protocmp"
+	"gotest.tools/v3/assert"
+)
+
+func TestValidateParent(t *testing.T) {
+	t.Parallel()
+	for _, tt := range []struct {
+		name   string
+		server mockFreightServer
+		f      func(*testing.T, context.Context, *mockFreightServer, *grpc.ClientConn)
+	}{
+		{
+			name: "non-parent request",
+			server: mockFreightServer{
+				getShipperResponse: &examplefreightv1.Shipper{Name: "shippers/1"},
+			},
+			f: func(t *testing.T, ctx context.Context, server *mockFreightServer, conn *grpc.ClientConn) {
+				client := examplefreightv1.NewFreightServiceClient(conn)
+				response, err := client.GetShipper(ctx, &examplefreightv1.GetShipperRequest{
+					Name: "shippers/1",
+				})
+				assert.DeepEqual(t, server.getShipperResponse, response, protocmp.Transform())
+				assert.NilError(t, err)
+			},
+		},
+
+		{
+			name: "valid parent response",
+			server: mockFreightServer{
+				listSitesResponse: &examplefreightv1.ListSitesResponse{
+					Sites: []*examplefreightv1.Site{
+						{Name: "shippers/1/sites/1"},
+						{Name: "shippers/1/sites/2"},
+						{Name: "shippers/1/sites/3"},
+					},
+				},
+			},
+			f: func(t *testing.T, ctx context.Context, server *mockFreightServer, conn *grpc.ClientConn) {
+				client := examplefreightv1.NewFreightServiceClient(conn)
+				response, err := client.ListSites(ctx, &examplefreightv1.ListSitesRequest{
+					Parent: "shippers/1",
+				})
+				assert.DeepEqual(t, server.listSitesResponse, response, protocmp.Transform())
+				assert.NilError(t, err)
+			},
+		},
+
+		{
+			name: "valid wildcard parent response",
+			server: mockFreightServer{
+				listSitesResponse: &examplefreightv1.ListSitesResponse{
+					Sites: []*examplefreightv1.Site{
+						{Name: "shippers/1/sites/1"},
+						{Name: "shippers/1/sites/2"},
```

Comment:
Make sure we properly handle multi-parent response.
```suggestion
						{Name: "shippers/2/sites/2"},
```

Reply:
Good point! Fixing

Reply:
Fixed